node/src/crypto/error.rs

```diff
@@ -53,6 +54,10 @@ pub enum Error {
     /// Error trying to manipulate the system key.
     #[error("invalid operation on system key: {0}")]
     System(String),
+
+    /// Error related to the underlying signature crate
+    #[error("Error in signature")]
+    Signature(SignatureError),
```

Comment:
@Fraser999 With this error message @darthsiroftardis isn't leaking any secrets in the Display impl.  Do we want to have a custom Debug impl too?

Reply:
We have this other error variant that might be problematic:

```rust
    /// Pem format error.
    #[error("pem error: {0}")]
    FromPem(String),
```

https://github.com/CasperLabs/casper-node/pull/1228/files#diff-c23caa49a8f80268f2deff3f2e467bdf5d855e1784c2927151f040c0018caeedR40

@Fraser999 I propose getting rid of these error variants that have arbitrary strings as payloads (in a separate story), as they are security hazards.

Reply:
Chatting with @Fraser999, we are thinking that the `Signature` variant here should be broken up into Ed25519 and Secp256k1 variants and we should get rid of the `From` impl.  This way we can better debug.  We'll earmark this for follow up work.